lib/liquid-fixpoint/src/lib.rs

```diff
@@ -153,9 +153,11 @@ pub struct ConstDecl<T: Types> {
 #[derive_where(Hash)]
 pub struct FunDef<T: Types> {
     pub name: T::Var,
+    pub arity: usize,
     pub args: Vec<(T::Var, Sort<T>)>,
     pub out: Sort<T>,
-    pub body: Expr<T>,
+    // Optional body for uninterpreted ("hide") functions
+    pub body: Option<Expr<T>>,
```

Comment:
We already have `ConstDecl` for `uif`. We should decide whether to emit a const or a fun def in `fixpoint_encoding`

Reply:
Yes, good point! But note that this will require beefing up the `DataSort` currently used by `ConstDecl` to include `func` sorts.

Reply:
But it already does, no? We are already using it for uifs. See `register_uif`

Reply:
NVM ignore above